src/lib/support/logging/CHIPLogging.cpp

```diff
@@ -206,63 +193,64 @@ DLL_EXPORT void LogByteSpan(uint8_t module, uint8_t category, const chip::ByteSp
 
 void LogV(uint8_t module, uint8_t category, const char * msg, va_list args)
 {
-    if (!IsCategoryEnabled(category))
-    {
-        return;
-    }
-
-    char moduleName[chip::Logging::kMaxModuleNameLen + 1];
-    GetModuleName(moduleName, module);
-
     LogRedirectCallback_t redirect = sLogRedirectCallback.load();
-
     if (redirect != nullptr)
     {
-        redirect(moduleName, category, msg, args);
+        redirect(GetModuleName(static_cast<LogModule>(module)), category, msg, args);
     }
+#if !CHIP_PLATFORM_CUSTOM_LOG
```

Comment:
We spoke about this in the other review, but for the case where we have a redirected call, did you want to log there as well as using the platform logger as well?

That is, should we remove the else statement and always log using the platform logger?

Reply:
Maybe `CHIP_PLATFORM_CUSTOM_LOG` is not named particularly well... if it is set, the platform log is hooked directly into the `ChipLog*` macros, so this code only has to handle the redirection part.

Reply:
I'm probably missing something. If the redirect is set by an app, it wouldn't hit the os_log at all though right? I thought we wanted to keep that around for all logs regardless of the app redirection being enabled? Or was that thought specifically only for MTR logs that you wanted to keep in os_logs (rather than all matter logs)?

Reply:
`os_log` comes in via `ChipPlatformLog` in `platform/Darwin/Logging.h`. It calls `os_log_with_type` before calling into the normal code path (`chip::Logging::Log` etc)

Reply:
New implementation is more straightforward, thanks!